namespace.go

```diff
@@ -0,0 +1,39 @@
+package vector
+
+import "errors"
+
+const deleteNamespacePath = "/delete-namespace"
+const listNamespacesPath = "/list-namespaces"
+
+// Namespace returns a new Index client for a specific namespace.
+func (ix *Index) Namespace(ns string) (i *Index) {
```

Comment:
A note: We might want to align this with the vector-js implementation. They are returning a `Namespace` object in this method, instead of `Index`. I am not sure which one is better, but I am inclined to returning a `Namespace` object which could be implemented as a wrapper around `Index`. This way, we can only expose per-namespace operations over this object, instead of both per-namespace and per-index operations (like only exporting upsert, and not info)

Reply:
✅